PublicDashboards: refactor service

evictorero opened the pull request:
**What this PR does / why we need it**:
* Removes `dashboard_query.go` files and package  and moves that logic into the service
* Splits `service` into `service.go` and `query.go` with tests into `service_test.go` and `query_test.go`
* Unexports methods not used outside the package
* Renames methods and functions for better legibility

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes https://github.com/grafana/grafana-partnerships-team/issues/436

evictorero committed:
PublicDashboards: refactor service

evictorero committed:
fix lint errors

evictorero committed:
Merge branch 'main' into evictorero/public-dashboards-service-refactor

owensmallwood reviewed (approved):
Looks good 👍

jalevin commented on pkg/services/publicdashboards/api/middleware.go:
```diff
@@ -50,7 +50,7 @@ func RequiresExistingAccessToken(publicDashboardService publicdashboards.Service
 
 		// Check that the access token references an enabled public dashboard
-		exists, err := publicDashboardService.AccessTokenExists(c.Req.Context(), accessToken)
+		exists, err := publicDashboardService.EnabledPublicDashboardExistsByAccessToken(c.Req.Context(), accessToken)
```
Yes! Thank you for tackling this.

What do you think about `PublicDashboardEnabledExistsByAccessToken` ? 

@owensmallwood thoughts?

evictorero replied:
I can change it but in that case I will change `EnabledPublicDashboardExistsByDashboardUid ` too.

jalevin replied:
Works for me!

jalevin reviewed (approved):
Looks good- one minor change on naming if we think necessary. Current name does not quite match the interface methods.

Also, it looks like we deleted 1679 lines and added 1605. Did we miss anything? I would expect this to be 1:1 code change

evictorero commented:
> e we deleted 1679 lines

That's a good point but I don't expect the same numbers because with code reorg there are duplicated imports, also I removed a func not being used and it test `HasExpressionQuery`.

evictorero committed:
renaming functions
